### PR TITLE
Tw 251/228: cli repo docs output

### PR DESCRIPTION
**Pull request by DJSanti**
**PREVIEW:** https://temporal-documentation-hnuk9f7tt.preview.thundergun.io/cli

## What does this PR do?
This PR allows the Docs repo to pull the CLI executable and generate the documentation provided by it.

## Notes to reviewers
Currently waiting on a new release in order to keep testing this function.

### CODE REVIEW

**Comment by DJSanti** [comment]
Note: I recently pushed a change within the CLI repository that will correct some issues I found in these documents.

**Comment by DJSanti** [comment]
I am aware of the broken links. Can fix or wait on them.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -154,6 +154,14 @@ This command runs snipsync, runs the Assembly Workflow, runs dPrint formatting,
 Any build issues are logged to your console.
 Fix all build issues before submitting a PR.
 
+## Retrieving documents from the CLI repository
+
+The `cli` repository contains a tool that automatically generates the reference documents for this feature. 
+These documents will be updated per CLI release.
+
+To get the documents, run `yarn assemble --cli`.
+Move the documents from /cli/docs to /cli, and delete any duplicate files.
```
[the remark this answers is not in the document]
Comment (DJSanti) [reply CONTRIBUTING.md:163]: fortunately, it does this now. I'll delete this passage.

**Comment by DJSanti** [comment]
@flossypurse Need to talk with you about changelog links and glossary. Those are the only hangups

**Review by flossypurse — COMMENTED** [review]
This MVP is looking pretty good.

Before we merge, we will need to add some messaging about which tool to use (tctl v1 vs temporal).

And let's get a stamp from @feedmeapples !

**Comment by bergundy** [comment]
- https://temporal-documentation-cevv7otib.preview.thundergun.io/cli/cmd-options - the options are command specific, there may be cases where the same option is used for different commands with different meaning.
- `temporal workflow trace` is not implemented and should be excluded from the public API and docs (cc @feedmeapples)
- For batch the description could use some work, there's more to it than `without having to repeat yourself on the command line`.

I can review more in depth later this week.

File: docs-src/cli/activity/fail.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+description: Fails an Activity.
+tags:
+	- cli
+---
+
+The `temporal activity fail` command fails an [Activity Execution](/concepts/what-is-an-activity-execution).
+
+Use the options listed below to change the behavior of this command.
+Make sure to write the command as follows:
+`temporal activity fail [command options] [arguments]`
```
Comment (bergundy) [inline docs-src/cli/activity/fail.md:14]: This command does not take any arguments, the comment is misleading.

File: docs-src/cli/activity/index.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+id: index
+title: temporal activity
+sidebar_label: activity
+description: Operations that can be performed on Workflow Activities.
+tags:
+	- cli
+---
+
+Activity commands enable operations on [Activity Executions](/concepts/what-is-an-activity-execution).
```
Comment (bergundy) [inline docs-src/cli/activity/index.md:10]: I didn't see this in rendered.

Sentence could be improved:

```suggestion
Operate on [Activity Executions](/concepts/what-is-an-activity-execution).
```

**Comment by bergundy** [comment]
The command line options should be inlined to avoid losing context when browsing the reference.
We should show the option groups like we do in the CLI output.

File: docs-src/cli/batch/describe.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+description: Describe a Batch operation job.
+tags:
+	- cli
+---
+
+The `temporal batch describe` command shows the progress of an ongoing Batch job.
+
+Use the command options listed below to change the information returned by this command.
+Make sure to write the command in this format:
+`temporal batch describe [command options] [arguments]`
```
Comment (bergundy) [inline docs-src/cli/batch/describe.md:14]: This doesn't take any arguments either.

File: docs-src/cli/batch/index.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+	- cli
+---
+
+Batch commands allow you to change multiple [Workflow Executions](/concepts/what-is-a-workflow-execution) without having to repeat yourself on the command line.
+In order to do this, you provide the command with a [List Filter](/concepts/what-is-visibility) and the type of Batch job to execute.
+
+The List Filter identifies the Workflow Executions that will be affected by the Batch job.
+The Batch type determines the other parameters that need to be provided, along with what is being affected on the Workflow Executions.
+
+To start the Batch job, run `temporal workflow query`.
```
Comment (bergundy) [inline docs-src/cli/batch/index.md:16]: That's not how you start a batch job. Could use a concrete example.

**Comment by bergundy** [comment]
Generally we could use more examples instead a generic `command [options] [arguments]` message.

**Comment by bergundy** [comment]
See the schedule docs for how a good example helps.

https://temporal-documentation-cevv7otib.preview.thundergun.io/cli/schedule

I wrote this and noticed that the examples are out of date @feedmeapples.

**Comment by bergundy** [comment]
@DJSanti you shouldn't update the text here, it should be in the CLI repo:

https://github.com/temporalio/cli/blob/6e237fcc332a0bf531bcbebe33c4d1453325b5c6/common/defs-cmds.go#L226

**Comment by bergundy** [comment]
Also, please don't use markdown in strings rendered by the CLI.

File: docs-src/cli/batch/index.md (hunk 5 above)
Comment (DJSanti) [reply docs-src/cli/batch/index.md:16]: I am generating what is provided in the CLI tool. Do you have a concrete example I could reference?

File: docs-src/cli/activity/fail.md (hunk 2 above)
Comment (DJSanti) [reply docs-src/cli/activity/fail.md:14]: Noted. Will revise CLI text.

**Comment by DJSanti** [comment]
Just pushed a custom help template that should solve the problem of the wrong text being sourced for Usage in CLI.

**Comment by DJSanti** [comment]
@bergundy I will review and itemize the feedback. So far, looks like we need updated examples in the text. I just need to know if this is across the board.

**Comment by DJSanti** [comment]
Changes merged in CLI repo

**Review by flossypurse — APPROVED** [review]

**Pull request merged**